library/ftd_install.py

```diff
@@ -0,0 +1,265 @@
+                                hostname=hostname)
+    finally:
+        dev.disconnect()
+
+
+PLATFORM_TO_INSTALL_MAP = {
+    'Cisco ASA5516-X Threat Defense': provision_ftd_5500x_with_kenton_platform
+}
+
+
+def main():
+    fields = dict(
+        device_hostname=dict(type='str', required=True),
+        device_password=dict(type='str', required=True, no_log=True),
+        device_sudo_password=dict(type='str', required=False, no_log=True),
+        device_ip=dict(type='str', required=False),
+        device_netmask=dict(type='str', required=False),
+        device_gateway=dict(type='str', required=False),
+        dns_server=dict(type='str', required=False),
+
+        console_ip=dict(type='str', required=True),
+        console_port=dict(type='str', required=True),
+        console_username=dict(type='str', required=True),
+        console_password=dict(type='str', required=True, no_log=True),
+
+        tftp_server=dict(type='str', required=True),
+        rommon_file_location=dict(type='str', required=True),
+        image_file_location=dict(type='str', required=True),
+        image_version=dict(type='str', required=False)
+    )
+    module = AnsibleModule(argument_spec=fields)
+    if not HAS_KICK:
+        module.fail_json(msg='Kick Python module is required to run this module.')
+
+    connection = Connection(module._socket_path)
+    resource = BaseConfigurationResource(connection, module.check_mode)
+
+    system_info = get_system_info(resource)
+    check_that_model_is_supported(module, system_info)
+    check_that_update_is_needed(module, system_info)
+    check_management_and_dns_params(resource, module.params)
+
+    provision_method = PLATFORM_TO_INSTALL_MAP[system_info['platformModel']]
+    provision_method(module.params)
+
+    module.exit_json(changed=True, result=module.params)
+
+
+def get_system_info(resource):
+    path_params = {ParamName.PATH_PARAMS: PATH_PARAMS_FOR_DEFAULT_OBJ}
+    system_info = resource.execute_operation(GET_SYSTEM_INFO_OPERATION, path_params)
+    return system_info
+
+
+def check_that_model_is_supported(module, system_info):
+    platform_model = system_info['platformModel']
+    if platform_model not in PLATFORM_TO_INSTALL_MAP:
+        module.fail_json(msg="Platform '%s' is not supported by this module." % platform_model)
+
+
+def check_that_update_is_needed(module, system_info):
+    target_ftd_version = module.params["image_version"]
+    if target_ftd_version == system_info['softwareVersion']:
+        module.exit_json(changed=False, msg="FTD already has %s version of software installed." % target_ftd_version)
+
+
+def check_management_and_dns_params(resource, params):
+    if not params.get('device_ip') or not params.get('device_netmask') or not params.get('device_gateway'):
+        management_ip = resource.execute_operation(GET_MANAGEMENT_IP_LIST_OPERATION, {})['items'][0]
+        params['device_ip'] = params.get('device_ip') or management_ip['ipv4Address']
```

Comment:
can't we just pass default argument to **.get** method here?
`        params['device_ip'] = params.get('device_ip', management_ip['ipv4Address'])
`

Reply:
Ok. got it